### ISSUE TITLE
Dispatch instead of symbol checking

### ISSUE THREAD

**Issue by matbesancon** [anriseth/MultiJuMP.jl#23]
Related to the solverhook method doing dispatch manually over symbols, a simpler design could leverage Julia dispatch and types for methods:

```julia
abstract type MultiOptMethod end

"""
multi_solve(mt::MultiOptMethod) solves the multi-objective problem with method mt
"""
function multi_solve end

"""
epsilon-constraint method, parametrized over the epsilon value
"""
struct EpsMethod <: MultiOptMethod
   eps::Float64
end

mutable struct MultiData
# ...
mt::MultiOptMethod
end

function solvehook(m::Model,
                   inequalityconstraint::Bool = false, kwargs...)
# dispatches over method from MultiData
end
```

**Comment by anriseth** [comment]
Good idea. The original approach was pretty un-Julian :p

**Comment by matbesancon** [comment]
fixed by #27
